src/DistributedTracingManager.ts

```diff
@@ -0,0 +1,223 @@
+        }
+
+        if (samplingRate !== undefined) {
+            twinPatch.properties.desired[Constants.DISTRIBUTED_TWIN_NAME].sampling_rate = samplingRate;
+        }
+
+        return await this.updateDeviceTwinJob(twinPatch, iotHubConnectionString, deviceIds);
+    }
+
+    private updateDeviceTwinJob(twinPatch, iotHubConnectionString: string, deviceIds: string[]) {
+        return new Promise((resolve, reject) => {
+            let twinJobId = uuid.v4();
+            let jobClient = iothub.JobClient.fromConnectionString(iotHubConnectionString);
+
+            let queryCondition = this.generateQureyCondition(deviceIds);
+            let startTime = new Date();
+            let maxExecutionTimeInSeconds = 300;
+
+            jobClient.scheduleTwinUpdate(twinJobId,
+                queryCondition,
+                twinPatch,
+                startTime,
+                maxExecutionTimeInSeconds,
+                (err) => {
+                    if (err) {
+                        reject("Could not schedule distributed tracing setting update job: " + err.message);
+                    } else {
+                        this.monitorJob(twinJobId, jobClient, (e, result) => {
+                            if (e) {
+                                reject("Could not monitor distributed tracing setting update job: " + e.message);
+                            } else {
+                                resolve(JSON.stringify(result, null, 2));
+                            }
+                        });
+                    }
+                });
+        });
+    }
+
+    private generateQureyCondition(deviceids: string[]): string {
+        const deviceIdsWithQuotes = deviceids.map((id) => "'" + id + "'");
+        return `deviceId IN [${deviceIdsWithQuotes.join(",")}]`;
+    }
+
+    private monitorJob(jobId, jobClient, callback) {
+        let jobMonitorInterval = setInterval(() => {
+            jobClient.getJob(jobId, (err, result) => {
+                if (err) {
+                    callback(err);
+                } else {
+                    if (result.status === "completed" || result.status === "failed" || result.status === "cancelled") {
+                        clearInterval(jobMonitorInterval);
+                        callback(null, result);
+                    }
+                }
+            });
+        }, 1000);
+    }
+
+    private getSamplingModePickupItems(): SamplingModeItem[] {
+        return [true, false].map((samplingMode) => new SamplingModeItem(samplingMode));
+    }
+
+    private async promptForSamplingRate(prompt: string, defaultValue: number): Promise<number> {
+        if (defaultValue === undefined || defaultValue > 100 || defaultValue < 0) {
+            defaultValue = 100;
+        }
+        let samplingRate: string = await vscode.window.showInputBox({ prompt, value: defaultValue.toString(), ignoreFocusOut: true });
```

Comment:
For validation, please use `validateInput` method, refer to https://github.com/Microsoft/vscode-azure-iot-toolkit/blob/c9c7b713c438362f9d77fe6f6587ec09f4f62483/src/iotHubResourceExplorer.ts#L190

Reply:
fixed